src/components/ReanimatedSwipeable.tsx

```diff
@@ -333,37 +343,64 @@ const Swipeable = forwardRef<SwipeableMethods, SwipeableProps>(
               [1, 0, 0]
             )
           : 0;
-    };
+    }, [
+      appliedTranslation,
+      friction,
+      leftWidth.value,
+      overshootFriction,
+      overshootLeftProp,
+      overshootRightProp,
+      rightWidth.value,
+      rowState.value,
+      showLeftProgress,
+      showRightProgress,
+      userDrag.value,
+      updateRightElementWidth,
+    ]);
 
     const dispatchImmediateEvents = useCallback(
       (fromValue: number, toValue: number) => {
+        'worklet';
         if (toValue > 0) {
-          onSwipeableWillOpen?.(SwipeDirection.RIGHT);
+          onSwipeableWillOpen &&
```

Comment:
Why don't we move this check to the `if` above?

```tsx
if (toValue > 0 && onSwipeableWillOpen){
  runOnJS(onSwipeableWillOpen)(SwipeDirection.RIGHT);
}
```

Reply:
I agree, your approach looks cleaner.

Changed in [155f20f](https://github.com/software-mansion/react-native-gesture-handler/pull/3165/commits/155f20f8a78f0dfd503e4f47bf16f9c2553f2457)